### DISCUSSION TITLE
How can we download all dependencies to a local folder and install dependencies from a local folder?

### DISCUSSION THREAD

**Discussion by canol** [pdm-project/pdm#3249] [Q&A]
Hello, we are considering to switch to PDM instead of using regular pip & venv tools, but we are trying to come up with an "offline" workflow using PDM.

90% of the time we are connected to internet and PDM works well for those use cases. But in some cases we need to install dependencies of our projects on computers that don't have internet or network connections because the computers are located in highly secured areas (defense, radiation testing, clean rooms etc.).

Our current solution is to download all dependencies in requirements.txt to a local folder:

`pip download -r requirements.txt -d [PATH_TO_LOCAL_FOLDER]`

Put this folder to a USB flash disk, and then copy to the target computer. Then execute this command:

`pip install -r requirements.txt --no-index --find-links [PATH_TO_LOCAL_FOLDER]`

And this works very well. But is there a way to achieve this with PDM?

**Comment by monchin** [comment]
I know [this plugin](
https://github.com/pdm-project/pdm-download) but I've never used it. Perhaps you can have a try.
